### ISSUE TITLE
Enhancement: Jamm ESSID instead of BSSID 0r Multiple BSSIDs

### ISSUE THREAD

**Issue by kn0wledge** [wifiphisher/wifiphisher#510]
This would be in case you are trying to use wifiphisher in an environment with multiple wifi locations with the same ESSID. Now, when you jam one BSSID the devices would auto-hop to the next nearest AP. If you jam the ESSID then all of the APs would get jammed increasing success likelihood. 

Another option is being able to select to jam more than 1 BSSID. 

Another must have feature to improve the functionality of wifiphisher.

**Comment by minanagehsalalma** [comment]
@kn0wledge good idea

**Comment by minanagehsalalma** [comment]
@kn0wledge would karma attack work on wpa 2 now days

**Comment by alternatexp** [comment]
Again, this can be achieved with `mdk3` by using a blacklist.

I can't see how this is useful when you're using the same ESSID as the target AP, you'd be deauthing everyone on your fake AP as well.

**Comment by alternatexp** [comment]
@minanagehsalalma karma attacks are far better, require the same channel and BSSID (and I think encryption type), however, I have done successful karma attacks using wifiphisher as some devices don't seem to check the encryption type. Karma attacks require a very strong signal.

**Comment by alternatexp** [comment]
> Another option is being able to select to jam more than 1 BSSID.

I suggest ditching the Wifiphisher jamming feature and just using mdk3, you can already do whatever you want.

**Comment by kn0wledge** [comment]
@alternatexp for some reason, when I use mdk3 I'm not able to jam any device. I use the blacklist command (kali) but nothing happens . Any specific syntax for the txt file containing the bssids?

**Comment by minanagehsalalma** [comment]
@alternatexp i must  the wpa key to do a karma attack

**Comment by kn0wledge** [comment]
Good point about jamming your own if we use ESSID option , so I'm guessing mdk3 with white or blacklist is the only option for now . So since it exists, then my suggestion is to integrate it in the wifiphisher features (same as fluxion does to a certain extent).

**Comment by alternatexp** [comment]
@kn0wledge yes, I've requested the mdk3 feature to be part of Wifiphisher before, but using the -nJ option combined with mdk3 is not that big of a hassle.

> @alternatexp for some reason, when I use mdk3 I'm not able to jam any device.

Make sure you put your interface in monitor mode manually before you deauth:
(the following command also spoofs your MAC address)

`airmon-ng start wlan2 && ifconfig wlan2mon down && macchanger -r wlan2mon && ifconfig wlan2mon up && echo "Interface is ready"`

Replace **wlan2** with the interface you want to use for deauthing.

**Comment by alternatexp** [comment]
@minanagehsalalma said:

> @alternatexp i must the wpa key to do a karma attack ??

I'm not too sure to be honest, but definitely need the same BSSID, ESSID and channel (and probably encryption type, thats why I also think the same WPA key)

**Comment by minanagehsalalma** [comment]
@alternatexp ok...  what do you think  @kn0wledge  ???

**Comment by kn0wledge** [comment]
@minanagehsalalma I'm not sure either .. haven't tried it in a while .. the connections I get using Karma are usually those without a wpa key

**Comment by alternatexp** [comment]
Yeah, WPA is different

**Comment by minanagehsalalma** [comment]
@kn0wledge @alternatexp so karma attack will not be helpful in wifiphisher !!!!!

**Comment by alternatexp** [comment]
It could be a cool addition for WEP protected APs

**Comment by minanagehsalalma** [comment]
@alternatexp wep don't need five minutes to hack...  cause it to weak security...  youcan hack with aircrack in just 2 minutes

**Comment by alternatexp** [comment]
Karma attacks are great for 3G and 4G networks

**Comment by kn0wledge** [comment]
@blackHatMonkey I guess you can close the topic if MDK3 is the way to go . Thx

**Comment by blackHatMonkey** [comment]
@kn0wledge I'm not really sure about. @sophron might have a different opinion about it.

**Comment by sophron** [comment]
We need to rely on our own code for deauth attacks. Channel hopping is a feature that we want to have.

**Comment by minanagehsalalma** [comment]
@sophron i agree with you.

**Comment by alternatexp** [comment]
I completely disagree

@sophron said:

> We need to rely on our own code for deauth attacks. Channel hopping is a feature that we want to have.

What do you mean? That there is no room for mdk3 (which already comes preinstalled on all Kali distributions)?

Are you trying to say that wifiphisher doesn't **rely** on dnsmasq, PyRIC, tornado, blessings?

As I have pointed out with various issues (if not all deauth related issues), that mdk3 **is** the better option for various reasons (channel hopping, multiple BSSIDs, ...)

Again, there is no need to tie yourself to anything.

**Comment by sophron** [comment]
PyRIC, tornado, blessings are all Python libraries, not external tools.

Deauthentication is an important part for associating the victim with the rogue Access Point. We need to have our own code that we can easily extend and customize accordingly.

Of'course anyone is free to use the `--nojamming` option and use mdk3 instead.

**Comment by alternatexp** [comment]
@sophron fair enough, but mdk3 comes preinstalled on all distributions, even the Lite versions.

I will be using mdk3 for now, until the same features are present in Wifiphisher
